dotCMS/src/main/java/com/dotmarketing/util/XMLUtils.java

```diff
@@ -17,4 +28,76 @@ public static String xmlEscape(String str) {
 	        return org.apache.commons.lang.StringEscapeUtils.escapeXml(str);
 	   
 	}
+	
+    public static final String prologue = "<?xml version=\"1.1\" encoding=\"UTF-8\"?>\n";
+	
+	public static boolean needsPrologue(File file) {
+	    return !readFirstLine(file).startsWith("<?xml version=");
+	}
+	
+    public static boolean addPrologueIfNeeded(final File file)  {
+        if(!needsPrologue(file)) {
+            return false;
+        }
+        Logger.info(XMLUtils.class, "prepending XML 1.1 prologue  to :" + file);
+        final File tempFile = new File(file.getParentFile(), "prologue" + file.getName());
+        try(InputStream input = new SequenceInputStream(new ByteArrayInputStream(prologue.getBytes("UTF-8")), Files.newInputStream(file.toPath())); OutputStream output = Files.newOutputStream(tempFile.toPath()) ){
+            IOUtils.copy(input, output );
+        }
+        catch(Exception e) {
+            throw new DotRuntimeException(e);
+        }
+        if(tempFile.length()<= file.length()) {
+            Logger.warn(XMLUtils.class, "unable to add prologue to xml file");
+            tempFile.delete();
+            return false;
+        }
+        return tempFile.renameTo(file);
+        
+        
+    }
+	
+	
+    public static String toXML10(final String input) {
+        return xml10pattern.matcher(input).replaceAll("");
+        
+        
+    }
+    
+    public static String toXML11(final String input) {
+        return xml11pattern.matcher(input).replaceAll("");
+        
+        
+    }
+    
+	
+	
+	private static String readFirstLine(File file) {

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'file' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)